src/common/components/dateInput/DateInputGroup.tsx

```diff
@@ -15,19 +15,25 @@ interface Props {
 
 function DateGroup({ children, id = INPUT_GROUP_ID, label, error }: Props) {
   const isInvalid = Boolean(error);
+  const errorId = `${id}-error`;
 
   return (
     <div
       role="group"
       aria-labelledby={id}
+      aria-describedby={errorId}
       className={className({
         'hds-text-input': true,
         'hds-text-input--invalid': isInvalid,
       })}
     >
       <DateInputLabel id={id}>{label}</DateInputLabel>
       {children}
-      <span className="hds-text-input__helper-text">{error}</span>
+      {isInvalid && (
```

Comment:
The new error styles contain an icon so the error element has to be hidden until there's an error.